### PR TITLE
Update rustc_pattern_analysis

### PR DESCRIPTION
**Pull request by Nadrieril**
This should fix https://github.com/rust-lang/rust-analyzer/issues/16656 but I can't check because we don't have a reproducer.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -88,7 +88,7 @@ ra-ap-rustc_lexer = { version = "0.35.0", default-features = false }
 ra-ap-rustc_parse_format = { version = "0.35.0", default-features = false }
 ra-ap-rustc_index = { version = "0.35.0", default-features = false }
 ra-ap-rustc_abi = { version = "0.35.0", default-features = false }
-ra-ap-rustc_pattern_analysis = { version = "0.37.0", default-features = false }
+ra-ap-rustc_pattern_analysis = { version = "0.40.0", default-features = false }
```
Comment (lnicola) [inline Cargo.toml:91]:
```suggestion
ra-ap-rustc_pattern_analysis = { version = "0.41.0", default-features = false }
```

I don't think 0.40 actually includes your fix, and 0.41 will hopefully come out in a couple of hours.

Comment (Nadrieril) [reply Cargo.toml:91]: I checked, it does include the fix :) It was published shortly after the fix landed, to my own surprise

Comment (lnicola) [reply Cargo.toml:91]: Ah, sorry then. Still, can you bump it to the new version? It's out now.

**Comment by lnicola** [comment]
Eh, whatever. We can upgrade next time.

@bors r+

File: Cargo.toml (hunk 1 above)
Comment (Nadrieril) [reply Cargo.toml:91]: Just saw that ^^ Yeah I'll bump next time it's useful

**Pull request merged**